### PR TITLE
Only include mobile reports from accessible apps in web apps restores

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

I tried to minimize changes as much as possible to isolate the intended change, which is to limit the apps used when building reports in restores. Generating mobile reports is an expensive part of a restore, so limiting the reports built to the apps that the restoring user has access to is a straightforward way to reduce the number of reports included in the restore.

I plan to add tests to the new filter utility as well, but wanted to put up a draft of the PR in case others had suggestions on better ways to achieve this. I'm pretty sure that the current implementation is sufficient to ensure that when a user's access to a specific app changes, those changes will be reflected in the next sync. If the user is added to an app, that app will be included in the list of apps referenced when collecting reports to sync. On the flip side, if a user's access is revoked, that app will not be included, and if a previous restore state exists, the relevant report fixture ids will be purged from the user's db.

### Rollback instructions

- [ ] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [ ] Risk label is set correctly
- [ ] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/utils.py [hunk 1]
```diff
@@ -38,6 +38,18 @@ def webapps_module(domain, app_id, module_id):
     return _webapps_url(domain, app_id, selections=[module_id])
 
 
+def filter_available_web_apps(apps, domain, user, is_preview):
```
Comment (esoergel) [inline corehq/apps/cloudcare/utils.py:41]: nice, I was thinking it'd be helpful to pull this out for easier testing

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 2]
```diff
@@ -109,11 +109,15 @@ def should_sync(self, restore_state):
...
         app_aware_sync_app = restore_state.params.app
+        web_apps_restore = restore_state.params.device_id.startswith('WebAppsLogin')
 
         if app_aware_sync_app:
             apps = [app_aware_sync_app]
         else:
             apps = get_apps_in_domain(restore_user.domain, include_remote=False)
+            if web_apps_restore and toggles.RESTORE_ACCESSIBLE_REPORTS_ONLY:
+                # use couch_user to determine app access
+                apps = filter_available_web_apps(apps, restore_user.domain, restore_user._couch_user, False)
```
Comment (esoergel) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py:120]: nit:
```suggestion
                apps = filter_available_web_apps(apps, restore_user.domain, restore_user._couch_user, is_preview=False)
```

File: corehq/apps/app_manager/fixtures/mobile_ucr.py [hunk 3]
```diff
@@ -109,11 +109,15 @@ def should_sync(self, restore_state):
 
     def _get_apps(self, restore_state, restore_user):
         app_aware_sync_app = restore_state.params.app
+        web_apps_restore = restore_state.params.device_id.startswith('WebAppsLogin')
```
Comment (esoergel) [inline corehq/apps/app_manager/fixtures/mobile_ucr.py:112]: Looks like there's a util for this:
https://github.com/dimagi/commcare-hq/blob/ae87eb2ebdd1a06437438327ebd02dfa913d6f83/corehq/ex-submodules/casexml/apps/phone/restore.py#L311-L313
```suggestion
        web_apps_restore = restore_state.params.is_webapps
```

**Review by esoergel — CHANGES_REQUESTED** [review]
I don't know whether `device_id` can ever be `None`, but since there's an existing util that guards against that condition, I'll say switching to that is blocking (otherwise that line might 500).  Otherwise this looks g2g

File: corehq/apps/cloudcare/utils.py (hunk 1 above)
Comment (gherceg) [reply corehq/apps/cloudcare/utils.py:41]: yeah I'm attempting to write tests for this now

File: corehq/apps/app_manager/fixtures/mobile_ucr.py (hunk 3 above)
Comment (gherceg) [reply corehq/apps/app_manager/fixtures/mobile_ucr.py]: See e76249bb00aa8db537b28f7e741550d0d8a029be

File: corehq/apps/app_manager/fixtures/mobile_ucr.py (hunk 2 above)
Comment (gherceg) [reply corehq/apps/app_manager/fixtures/mobile_ucr.py]: See e76249bb00aa8db537b28f7e741550d0d8a029be

**Comment by gherceg** [comment]
Likely going to close this in favor of a new PR that builds off of https://github.com/dimagi/commcare-hq/pull/34546, as this will allow for a cleaner implementation in mobile ucr code than what it is currently.

**Comment by gherceg** [comment]
Closing in favor of https://github.com/dimagi/commcare-hq/pull/34547

**Pull request closed without merging**